src/metabase/driver/clickhouse.clj

```diff
@@ -0,0 +1,201 @@
+(ns metabase.driver.clickhouse
+  (:require [clojure
+             [set :as set]
+             [string :as string]]
+            [honeysql
+             [core :as hsql]
+             [helpers :as h]]
+            [metabase.driver :as driver]
+            [metabase.driver.common :as driver.common]
+            [metabase.driver.sql-jdbc
+             [common :as sql-jdbc.common]
+             [connection :as sql-jdbc.conn]
+             [sync :as sql-jdbc.sync]]
+            [metabase.driver.sql.query-processor :as sql.qp]
+            [metabase.mbql.util :as mbql.u]
+            [metabase.models.field :as field]
+            [metabase.util :as u]
+            [metabase.util
+             [honeysql-extensions :as hx]
+             [ssh :as ssh]]))
+
+(driver/register! :clickhouse, :parent :sql-jdbc)
+
+(def ^:private default-base-types
```

Comment:
clickhouse also supports `Decimal` https://clickhouse.yandex/docs/ru/data_types/decimal/

Reply:
Thank you! Added the data type to the test.